Slideshow on front page of vivo

mrpatrickwest opened the pull request:
based on dco-viz repository. Still need to link to vivo pages for the larger visualizations instead of to the /dco-viz directory under apache root

Commit:
Slideshow on front page of vivo

based on dco-viz repository. Still need to link to vivo pages for the larger visualizations instead of to the /dco-viz directory under apache root

Commit:
Eliminating gap before footer

The min height for the main part of the page was set to 710px. Reduced this so that it doesn't have so much spacing before the footer.

mrpatrickwest commented:
needs some styling changes for the pie charts because the titles don't completely display.

Commit:
Works in Chrome and Safari

Commit:
slide show works in IE, Safari and Chrome

Pie charts still don't show up in firefox and there's a flash before the visualization renders.

Commit:
Pie hart as first slide works in firefox

The second one still doesn't display, but good to have in first slot anyway.

Commit:
Caption is now 11pt instead of 11px

px made it really, really small

am-e committed:
Added slideshow to page-home.ftl

Commit:
Merge branch 'master' into slideshow

Conflicts:
	themes/dco/templates/page-home.ftl

am-e committed:
Merged stats with slideshow

zhongh commented:
@am-e Did you put the newest update on https://dco.tw.rpi.edu/vivo/? It just looks nice on there.

am-e committed:
slideshow/stats working on Jquery 1.12.2, reordered slides

am-e committed:
cleanup of captions

zhongh commented:
I think it looks beautiful 👍

olyerickson commented:
The slideshow at https://dco.tw.rpi.edu/vivo/ looks fantastic; great job!

Commit:
Merge branch 'slideshow'

of https://github.com/tetherless-world/dco-vivo into slideshow

Conflicts:
	themes/dco/templates/page-home.ftl

Commit:
Pulled latest and merged with what we're seeing

There was code to pull for this branch and it conflicted with what was already here. Copied what we're seeing on dco.tw to resolve conflict.

mrpatrickwest commented:
what is here is what we're seeing on https://dco.tw.rpi.edu/vivo/ so should be current.

Commit:
Moving slideshow javascript and css to vivo theme

Instead of being separate repository make it part of vivo since the visualizations are in VIVO.

mrpatrickwest commented:
Are we okay with the new changes? Putting the js and css under VIVO instead of a separate repository? And should I remove the corresponding files from the dco-viz repository? Or keep them there for development purposes?

zednis commented:
I don't see a reason to delete any files from the dco-viz repository, I just think that repository should be deprecated at some point.
- Are all CSS and JS resources for the slideshow now in the dco-vivo:slideshow branch?
- Why have themes/dco/js/subjectAreas.js as a separate file?  All it does is define a JS string variable
- Why do we have 2 different versions of jquery in /js?  jquery-1.7.2.min.js and jquery-2.1.4.js
- **perhaps put the slideshow FTL into its own FTL that is imported by the homepage FTL instead of embedded directly?**

am-e commented:
- We do not need subjectAreas.js, It's a remnant of the older solution.
- As for the 2 different jquery versions, I had implemented the slideshow using jquery 2.1.4 but Hao's code for the statistics "requires" 1.7.2. I think we can just use 1.7.2, although I'd hoped we'd be able to use the latest.
- It would probably be a very smart move if we could create an FTL for the slideshow. 

Thanks Patrick for re-organizing the files into VIVO.

mrpatrickwest commented on themes/dco/js/expertiseCountQuery.js:
```diff
@@ -0,0 +1,53 @@
+/**
```
do we need expertiseCountQuery.js? Seems same as subjectAreas.js, just a single variable defined.

am-e replied:
We do not need both. They're both how things were done before the static files.

mrpatrickwest commented:
Can't we just link to an external jquery-1.7.2.min.js file instead of having our own? Or is it better to have our own?

Commit:
No longer using these javascript files

We switched to using jquery-1.7.2, so no longer need jquery-2.1.4

Don't need subjectAreas.js as it is just defining a single variable.

Commit:
No longer needing this, same as subjectAreas.js

This defined just a single variable that we no longer need.

zednis commented:
@mrpatrickwest I think it makes sense to refer to a public CDN for jquery (any version).

am-e commented:
In my experience https://cdnjs.cloudflare.com/ajax/libs/jquery/1.7.2/jquery.min.js is perfectly fine.

mrpatrickwest commented:
@zednis @am-e I hesitate separating out the slideshow (and stats) from page-home.ftl because of how difficult it was to get everything in the right order and to be put in the right place. I would like to go ahead and push what we have and in the future we could separate it out when we have more time. We've already spent way too much time on this and need to get going on other tasks that are far more important.

I know we had to move the css for the slideshow above the javascript for some weird reason. And we would need to link to the js for the stats and the css for the slideshow in page-home.ftl.

I'm sure it probably would be "easy" to do ... but not right now. We can create another ticket to separate it out and if someone has any spare time to work on it they are more than welcome to.

am-e commented:
Yes, I forgot the fact that getting the layout to fit and work well was hassle. Agreed.

zednis commented:
Not addressing this now will just make it technical debt we will likely not address for a long time (if ever).

If you feel strongly about pushing this in the current state then create a backlog ticket to refactor and please put near the top of the queue.

My preference would be to not push this until after the refactor.

Commit:
Separating out slideshow and stats from page-home

Instead of having it all in one place separated it out into two separate ftl and including from page-home.ftl

mrpatrickwest commented:
that went a lot smoother than I thought it would.

Need to create a few tickets here in that there appears to be a way to set scripts and styles to be included in a list somewhere and have them loaded from that list instead of hardcoding it into the page-home.ftl.

mrpatrickwest commented:
Can you both test again. Need to make sure this works in all the browsers

am-e commented:
We're talking on dco.tw? I checked in Chrome, Firefox and Safari.
I caught one issue, in the alert when "more information" is clicked on the Members pie, the message contains ".. and undefined publications" I say since it is the members pie we take out the second part of the sentence. That is what I did in the publications pie.
Otherwise, looks good.

zednis commented:
:+1: to the refactored slideshow CSS/JS

I tested the dco.tw main page slideshow with Chrome, Safari, and Firefox on OSX and it seemed to work in all.

There were some 404 errors in the web console:

> https://dco.tw.rpi.edu/dco-viz/css/result-light.css Failed to load resource: the server responded with a status of 404 (Not Found)
> jquery.min.js:4 GET https://dco.tw.rpi.edu/vivo/dco-viz/statsJSON/communityMemberCounts2.json 404 (Not Found)

Commit:
Pulling status from correct URL

Was grabbing one of the stats for a pie chart from /vivo/dco-viz instead of /dco-viz

mrpatrickwest commented:
@zednis fixed the two problems you mentioned above. I get no errors or warnings in console now for any of the pages.

I say :shipit:

zednis commented:
Looks good.  dco.tw.rpi.edu is running fast and I can confirm I see no errors.

:shipit:

am-e commented:
All looks good even on the phone.
On Tue, May 31, 2016 at 12:24 PM Stephan Zednik notifications@github.com
wrote:

> Looks good. dco.tw.rpi.edu is running fast and I can confirm I see no
> errors.
> 
> [image: :shipit:]
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> https://github.com/tetherless-world/dco-vivo/pull/45#issuecomment-222742426,
> or mute the thread
> https://github.com/notifications/unsubscribe/AIiVGHMRrfOLRaaS6AGQ1SiwAiHQ2Cy4ks5qHGDSgaJpZM4IZ1yY
> .
